zhmccli/_cmd_imageprofile.py

```diff
@@ -0,0 +1,1918 @@
+        'absolute-ziip-capping': None,
+        'dedicated-ziip-processors': 'number-dedicated-ziip-processors',
+        'reserved-dedicated-ziip-processors':
+            'number-reserved-dedicated-ziip-processors',
+        'shared-icf-processors': 'number-shared-icf-processors',
+        'reserved-shared-icf-processors':
+            'number-reserved-shared-icf-processors',
+        'initial-icf-processing-weight':
+            'initial-internal-cf-processing-weight',
+        'minimum-icf-processing-weight':
+            'minimum-internal-cf-processing-weight',
+        'maximum-icf-processing-weight':
+            'maximum-internal-cf-processing-weight',
+        'icf-processing-weight-capped':
+            'initial-internal-cf-processing-weight-capped',
+        'absolute-icf-capping': None,
+        'dedicated-icf-processors': 'number-dedicated-icf-processors',
+        'reserved-dedicated-icf-processors':
+            'number-reserved-dedicated-icf-processors',
+        'central-storage-origin': 'user-specified-central-storage-origin',
+        'access-basic-cpu-counter': 'basic-cpu-counter-authorization-control',
+        'access-problem-state-cpu-counter':
+            'problem-state-cpu-counter-authorization-control',
+        'access-crypto-activity-cpu-counter':
+            'crypto-activity-cpu-counter-authorization-control',
+        'access-extended-cpu-counter':
+            'extended-cpu-counter-authorization-control',
+        'access-coprocessor-cpu-counter':
+            'coprocessor-cpu-counter-authorization-control',
+        'access-diagnostic-sampling':
+            'diagnostic-sampling-authorization-control',
+        'access-basic-cpu-sampling': 'basic-cpu-sampling-authorization-control',
+        'permit-aes-key-import': 'permit-aes-key-import-functions',
+        'permit-des-key-import': 'permit-des-key-import-functions',
+        'permit-ecc-key-import': 'permit-ecc-key-import-functions',
+        'access-global-performance-data':
+            'global-performance-data-authorization-control',
+        'permit-global-iocds': 'io-configuration-authorization-control',
+        'permit-cross-partition-authority':
+            'cross-partition-authority-authorization-control',
+        'permit-bcpii-send-commands': 'security-bcpii-send-commands',
+        'permit-bcpii-receive-commands': 'security-bcpii-receive-commands',
+        'bcpii-receive-partitions': 'security-bcpii-receive-partition-list',
+        'time-offset-direction': 'time-offset-increase-decrease',
+    }
+
+    org_options = original_options(options)
+    if org_options['copy-name'] is not None:
+        copyprofile = find_imageprofile(
+            cmd_ctx, client, cpc_name, org_options['copy-name'])
+        properties = copy.deepcopy(copyprofile.properties)
+    else:
+        properties = {}
+    properties.update(options_to_properties(org_options, name_map))
+
+    # Specially handled options
+
+    if org_options['disk-partition-id'] == 'auto':
```

Comment:
So you suggest to change this:
```
    if org_options['disk-partition-id'] == 'auto':
        properties['disk-partition-id-automatic'] = True
        # 'disk-partition-id will be unchanged
    elif org_options['disk-partition-id'] is not None:
        properties['disk-partition-id-automatic'] = False
        properties['disk-partition-id'] = str2int(
            cmd_ctx, 'disk-partition-id', org_options['disk-partition-id'])
```
into something like this:
```
    set_property_disk_partition_id_and_friends(cmd_ctx, org_options, properties)
```
just because it occurs two times?

OK, two times in the image profile module, and two times in the load profile module. So now we talk about module-independent helper functions that are completely specific to these two modules.

Reply:
I found a way to remove the duplication, by moving the whole code for handling special options into one function that is used by create and update.